[change] Remove unneeded ssl package implementation definitions

feckert opened the pull request:
@nemesisdesign @okraits 
Which ssl implementation is used by the `openwisp-config` agent is decided in the curl package. Currently, the [wolfssl](https://github.com/openwrt/packages/blob/master/net/curl/Config.in#L7) implementation is used here.

The openwisp-agent is not compiled servivce. Therefore, the division into different packages is **not** necessary.
This commit removes all ssl package versions. There is now only **one** `openwisp-config` package.

This change is a result of the desire to integrate the openwisp-config as luci-app-openwisp into the LuCI repository.
https://github.com/openwrt/luci/pull/5471

PS:
openwisp monitoring is also affected by that change https://github.com/openwisp/openwrt-openwisp-monitoring/pull/78

nemesisdesign reviewed (commented):
This would be a nice simplification!

We'll have to update the README and the docs too.

I guess we have to recommend users to also installa an SSL library or the curl requests to any SSL HTTP endpoint will fail, or is SSL enabled by default now on OpenWrt?

feckert commented:
> We'll have to update the README and the docs too.

I can do that


> I guess we have to recommend users to also installa an SSL library or the curl requests to any SSL HTTP endpoint will fail, or is SSL enabled by default now on OpenWrt?

@nemesisdesign From my point of view, that is not necessary. Since the ssl backend used for curl is set at compile time. And the current situation is that wolfssl is default used.
Please see https://github.com/openwrt/packages/blob/master/net/curl/Config.in#L7.

nemesisdesign commented:
> > We'll have to update the README and the docs too.
> 
> I can do that

Thank you very much.
 
> > I guess we have to recommend users to also installa an SSL library or the curl requests to any SSL HTTP endpoint will fail, or is SSL enabled by default now on OpenWrt?
> 
> @nemesisdesign From my point of view, that is not necessary. Since the ssl backend used for curl is set at compile time. And the current situation is that wolfssl is default used. Please see https://github.com/openwrt/packages/blob/master/net/curl/Config.in#L7.

Ok sounds good to me!

okraits commented on README.rst:
```diff
@@ -296,20 +295,13 @@ Path to an executable script that will be called after the registration is compl
-- **openwisp-config-nossl**: doesn't depend on any SSL library and doesn't install trusted CA certificates
-
-The following procedure illustrates how to compile all the *openwisp-config* variants and their dependencies:
+The following procedure illustrates how to compile all the *openwisp-config* and their dependencies:
```
Better:

`The following procedure illustrates how to compile *openwisp-config* and its dependencies:`

feckert commented:
@nemesisdesign 
I have updated the documentation, and added only the change that creates only one package.
I am now doing the same for openwisp-monitoring.

okraits commented on README.rst:
```diff
@@ -319,23 +311,20 @@ The following procedure illustrates how to compile all the *openwisp-config* var
 
 Alternatively, you can configure your build interactively with ``make menuconfig``, in this case
-you will need to select the *openwisp-config* variant by going to ``Administration > openwisp``:
+you will need to select the *openwisp-config* by going to ``Administration > openwisp``:
```
Better:

`you will need to select *openwisp-config* by going to ``Administration > openwisp``:`

feckert commented:
I also need to fix https://github.com/openwisp/openwisp-config/blob/master/runbuild

devkapilbansal commented on runbuild:
```diff
@@ -44,10 +44,7 @@ sed -i '/routing/d' feeds.conf
-  echo "CONFIG_PACKAGE_openwisp-config-mbedtls=y"
-  echo "CONFIG_PACKAGE_openwisp-config-wolfssl=y"
-  echo "CONFIG_PACKAGE_openwisp-config-nossl=y"
+  echo "CONFIG_PACKAGE_openwisp-config=y"
 } >>.config
```
We can remove braces now as there is only one package to be added
```suggestion
  echo "CONFIG_PACKAGE_openwisp-config=y" >>.config
```

devkapilbansal commented on openwisp-config/Makefile:
```diff
@@ -10,40 +10,29 @@ PKG_VERSION:=$(shell cat ../VERSION)
-	DEPENDS:=+curl +lua +libuci-lua +luafilesystem +luci-lib-nixio $(3)
-	VARIANT:=$(1)
+	DEPENDS:=+curl +lua +libuci-lua +luafilesystem +luci-lib-nixio +ca-certificates
+	PROVIDES:= openwisp-config-openssl openwisp-config-mbedtls openwisp-config-wolfssl openwisp-config-nossl
```
One stupid question
If we are removing definitions of these packages, then why they are mentioned here?

feckert replied:
I added this because there may be someone who has entered this as a dependency somewhere. But we can also delete it, I would even prefer that.

feckert replied:
@devkapilbansal Should I remove it, would only be for back value compatibility if someone refers to one of the packages that are now no longer available

nemesisdesign replied:
@feckert the upcoming release of openwisp will be contain a number of backward incompatible changes, so if we want to do this it may be good to take the chance and do it now, from then onwards we can refrain from doing so many other backward incompatible changes like we have been needing to do lately, what do you think?

feckert replied:
@nemesisdesign thansk for you feedback.
Should fit now

devkapilbansal reviewed (changes requested):
Looks good, will test it soon :smile:

devkapilbansal reviewed (commented):
Hi @feckert ,
<details>
<summary>I tried to install in on VM and got following error</summary>
root@08-00-27-E4-9C-09:/tmp# opkg install openwisp-config_0.6.0a-1_all.ipk 
Installing openwisp-config (0.6.0a-1) to root...
Installing ca-certificates (20200601-1) to root...
Downloading http://downloads.openwrt.org/releases/19.07.2/packages/x86_64/base/ca-certificates_20200601-1_all.ipk
Collected errors:
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/03179a64.0 to 'Staat_der_Nederlanden_EV_Root_CA.crt': I/O error.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/062cdee6.0 to 'GlobalSign_Root_CA_-_R3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/064e0aa9.0 to 'QuoVadis_Root_CA_2_G3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/06dc52d5.0 to 'SSL.com_EV_Root_Certification_Authority_RSA_R2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/080911ac.0 to 'QuoVadis_Root_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/09789157.0 to 'Starfield_Services_Root_Certificate_Authority_-_G2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/0a775a30.0 to 'GTS_Root_R3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/0b1b94ef.0 to 'CFCA_EV_ROOT.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/0bf05006.0 to 'SSL.com_Root_Certification_Authority_ECC.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/0c4c9b6c.0 to 'Global_Chambersign_Root_-_2008.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/0f5dc4f3.0 to 'UCA_Extended_Validation_Root.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/0f6fa695.0 to 'GDCA_TrustAUTH_R5_ROOT.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/1001acf7.0 to 'GTS_Root_R1.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/106f3e4d.0 to 'Entrust_Root_Certification_Authority_-_EC1.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/128805a3.0 to 'EE_Certification_Centre_Root_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/14bc7599.0 to 'emSign_ECC_Root_CA_-_G3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/1636090b.0 to 'Hellenic_Academic_and_Research_Institutions_RootCA_2011.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/18856ac4.0 to 'SecureSign_RootCA11.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/1d3472b9.0 to 'GlobalSign_ECC_Root_CA_-_R5.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/1e08bfd1.0 to 'IdenTrust_Public_Sector_Root_CA_1.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/1e09d511.0 to 'T-TeleSec_GlobalRoot_Class_2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/244b5494.0 to 'DigiCert_High_Assurance_EV_Root_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/2923b3f9.0 to 'emSign_Root_CA_-_G1.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/2ae6433e.0 to 'CA_Disig_Root_R2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/2b349938.0 to 'AffirmTrust_Commercial.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/2e5ac55d.0 to 'DST_Root_CA_X3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/32888f65.0 to 'Hellenic_Academic_and_Research_Institutions_RootCA_2015.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/349f2832.0 to 'EC-ACC.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/3513523f.0 to 'DigiCert_Global_Root_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/3bde41ac.0 to 'Autoridad_de_Certificacion_Firmaprofesional_CIF_A62634068.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/3e44d2f7.0 to 'TrustCor_RootCert_CA-2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/3e45d192.0 to 'Hongkong_Post_Root_CA_1.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/40193066.0 to 'Certum_Trusted_Network_CA_2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/4042bcee.0 to 'ISRG_Root_X1.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/40547a79.0 to 'COMODO_Certification_Authority.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/406c9bb1.0 to 'emSign_Root_CA_-_C1.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/4304c5e5.0 to 'Network_Solutions_Certificate_Authority.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/48bec511.0 to 'Certum_Trusted_Network_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/4a6481c9.0 to 'GlobalSign_Root_CA_-_R2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/4b718d9b.0 to 'emSign_ECC_Root_CA_-_C3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/4bfab552.0 to 'Starfield_Root_Certificate_Authority_-_G2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/4f316efb.0 to 'SwissSign_Gold_CA_-_G2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/5273a94c.0 to 'E-Tugra_Certification_Authority.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/5443e9e3.0 to 'T-TeleSec_GlobalRoot_Class_3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/54657681.0 to 'Buypass_Class_2_Root_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/57bcb2da.0 to 'SwissSign_Silver_CA_-_G2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/5a4d6896.0 to 'Staat_der_Nederlanden_Root_CA_-_G3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/5ad8a5d6.0 to 'GlobalSign_Root_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/5c44d531.0 to 'Staat_der_Nederlanden_Root_CA_-_G2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/5cd81ad7.0 to 'TeliaSonera_Root_CA_v1.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/5d3033c5.0 to 'TrustCor_RootCert_CA-1.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/5e98733a.0 to 'Entrust_Root_Certification_Authority_-_G4.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/5f15c80c.0 to 'TWCA_Global_Root_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/607986c7.0 to 'DigiCert_Global_Root_G2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/626dceaf.0 to 'GTS_Root_R2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/6410666e.0 to 'Taiwan_GRCA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/653b494a.0 to 'Baltimore_CyberTrust_Root.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/68dd7389.0 to 'Hongkong_Post_Root_CA_3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/6b99d060.0 to 'Entrust_Root_Certification_Authority.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/6d41d539.0 to 'Amazon_Root_CA_2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/6fa5da56.0 to 'SSL.com_Root_Certification_Authority_RSA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/706f604c.0 to 'XRamp_Global_CA_Root.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/749e9e03.0 to 'QuoVadis_Root_CA_1_G3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/75d1b2ed.0 to 'DigiCert_Trusted_Root_G4.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/76cb8f92.0 to 'Cybertrust_Global_Root.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/76faf6c0.0 to 'QuoVadis_Root_CA_3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/7719f463.0 to 'Hellenic_Academic_and_Research_Institutions_ECC_RootCA_2015.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/773e07ad.0 to 'OISTE_WISeKey_Global_Root_GC_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/7aaf71c0.0 to 'TrustCor_ECA-1.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/7f3d5d1d.0 to 'DigiCert_Assured_ID_Root_G3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/8160b96c.0 to 'Microsec_e-Szigno_Root_CA_2009.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/8867006a.0 to 'GeoTrust_Universal_CA_2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/8cb5ee0f.0 to 'Amazon_Root_CA_3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/8d86cdd1.0 to 'certSIGN_ROOT_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/930ac5d2.0 to 'Actalis_Authentication_Root_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/93bc0acc.0 to 'AffirmTrust_Networking.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/988a38cb.0 to 'NetLock_Arany_=Class_Gold=_Főtanúsítvány.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/9c2e7d30.0 to 'Sonera_Class_2_Root_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/9c8dfbd4.0 to 'AffirmTrust_Premium_ECC.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/9d04f354.0 to 'DigiCert_Assured_ID_Root_G2.crt': Read-only file system.
 * wfopen: /etc/ssl/certs/ACCVRAIZ1.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/AC_RAIZ_FNMT-RCM.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Actalis_Authentication_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/AffirmTrust_Commercial.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/AffirmTrust_Networking.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/AffirmTrust_Premium.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/AffirmTrust_Premium_ECC.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Amazon_Root_CA_1.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Amazon_Root_CA_2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Amazon_Root_CA_3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Amazon_Root_CA_4.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Atos_TrustedRoot_2011.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Autoridad_de_Certificacion_Firmaprofesional_CIF_A62634068.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Baltimore_CyberTrust_Root.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Buypass_Class_2_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Buypass_Class_3_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/CA_Disig_Root_R2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/CFCA_EV_ROOT.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/COMODO_Certification_Authority.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/COMODO_ECC_Certification_Authority.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/COMODO_RSA_Certification_Authority.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Certigna.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Certigna_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Certum_Trusted_Network_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Certum_Trusted_Network_CA_2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Chambers_of_Commerce_Root_-_2008.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Comodo_AAA_Services_root.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Cybertrust_Global_Root.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/D-TRUST_Root_Class_3_CA_2_2009.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/D-TRUST_Root_Class_3_CA_2_EV_2009.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/DST_Root_CA_X3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/DigiCert_Assured_ID_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/DigiCert_Assured_ID_Root_G2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/DigiCert_Assured_ID_Root_G3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/DigiCert_Global_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/DigiCert_Global_Root_G2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/DigiCert_Global_Root_G3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/DigiCert_High_Assurance_EV_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/DigiCert_Trusted_Root_G4.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/E-Tugra_Certification_Authority.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/EC-ACC.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/EE_Certification_Centre_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Entrust.net_Premium_2048_Secure_Server_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Entrust_Root_Certification_Authority.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Entrust_Root_Certification_Authority_-_EC1.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Entrust_Root_Certification_Authority_-_G2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Entrust_Root_Certification_Authority_-_G4.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/GDCA_TrustAUTH_R5_ROOT.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/GTS_Root_R1.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/GTS_Root_R2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/GTS_Root_R3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/GTS_Root_R4.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/GeoTrust_Universal_CA_2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/GlobalSign_ECC_Root_CA_-_R4.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/GlobalSign_ECC_Root_CA_-_R5.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/GlobalSign_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/GlobalSign_Root_CA_-_R2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/GlobalSign_Root_CA_-_R3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/GlobalSign_Root_CA_-_R6.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Global_Chambersign_Root_-_2008.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Go_Daddy_Class_2_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Go_Daddy_Root_Certificate_Authority_-_G2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Hellenic_Academic_and_Research_Institutions_ECC_RootCA_2015.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Hellenic_Academic_and_Research_Institutions_RootCA_2011.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Hellenic_Academic_and_Research_Institutions_RootCA_2015.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Hongkong_Post_Root_CA_1.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Hongkong_Post_Root_CA_3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/ISRG_Root_X1.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/IdenTrust_Commercial_Root_CA_1.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/IdenTrust_Public_Sector_Root_CA_1.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Izenpe.com.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/LuxTrust_Global_Root_2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Microsec_e-Szigno_Root_CA_2009.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/NetLock_Arany_=Class_Gold=_Főtanúsítvány.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Network_Solutions_Certificate_Authority.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/OISTE_WISeKey_Global_Root_GA_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/OISTE_WISeKey_Global_Root_GB_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/OISTE_WISeKey_Global_Root_GC_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/QuoVadis_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/QuoVadis_Root_CA_1_G3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/QuoVadis_Root_CA_2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/QuoVadis_Root_CA_2_G3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/QuoVadis_Root_CA_3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/QuoVadis_Root_CA_3_G3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/SSL.com_EV_Root_Certification_Authority_ECC.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/SSL.com_EV_Root_Certification_Authority_RSA_R2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/SSL.com_Root_Certification_Authority_ECC.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/SSL.com_Root_Certification_Authority_RSA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/SZAFIR_ROOT_CA2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/SecureSign_RootCA11.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/SecureTrust_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Secure_Global_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Security_Communication_RootCA2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Security_Communication_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Sonera_Class_2_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Staat_der_Nederlanden_EV_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Staat_der_Nederlanden_Root_CA_-_G2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Staat_der_Nederlanden_Root_CA_-_G3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Starfield_Class_2_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Starfield_Root_Certificate_Authority_-_G2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Starfield_Services_Root_Certificate_Authority_-_G2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/SwissSign_Gold_CA_-_G2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/SwissSign_Silver_CA_-_G2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/T-TeleSec_GlobalRoot_Class_2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/T-TeleSec_GlobalRoot_Class_3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/TUBITAK_Kamu_SM_SSL_Kok_Sertifikasi_-_Surum_1.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/TWCA_Global_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/TWCA_Root_Certification_Authority.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Taiwan_GRCA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/TeliaSonera_Root_CA_v1.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/TrustCor_ECA-1.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/TrustCor_RootCert_CA-1.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/TrustCor_RootCert_CA-2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Trustis_FPS_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/UCA_Extended_Validation_Root.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/UCA_Global_G2_Root.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/USERTrust_ECC_Certification_Authority.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/USERTrust_RSA_Certification_Authority.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Verisign_Class_3_Public_Primary_Certification_Authority_-_G3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/XRamp_Global_CA_Root.crt: Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/a3418fda.0 to 'GTS_Root_R4.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/a94d09e5.0 to 'ACCVRAIZ1.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/aee5f10d.0 to 'Entrust.net_Premium_2048_Secure_Server_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/b0e59380.0 to 'GlobalSign_ECC_Root_CA_-_R4.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/b1159c4c.0 to 'DigiCert_Assured_ID_Root_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/b1b8a7f3.0 to 'OISTE_WISeKey_Global_Root_GA_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/b66938e9.0 to 'Secure_Global_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/b727005e.0 to 'AffirmTrust_Premium.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/b7a5b843.0 to 'TWCA_Root_Certification_Authority.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/c01eb047.0 to 'UCA_Global_G2_Root.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/c0ff1f52.0 to 'Verisign_Class_3_Public_Primary_Certification_Authority_-_G3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/c28a8a30.0 to 'D-TRUST_Root_Class_3_CA_2_2009.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/c47d9980.0 to 'Chambers_of_Commerce_Root_-_2008.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/ca6e4ad9.0 to 'ePKI_Root_Certification_Authority.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/cbf06781.0 to 'Go_Daddy_Root_Certificate_Authority_-_G2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/cc450945.0 to 'Izenpe.com.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/cd58d51e.0 to 'Security_Communication_RootCA2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/cd8c0d63.0 to 'AC_RAIZ_FNMT-RCM.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/ce5e74ef.0 to 'Amazon_Root_CA_1.crt': Read-only file system.
 * wfopen: /etc/ssl/certs/certSIGN_ROOT_CA.crt: Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/d4dae3dd.0 to 'D-TRUST_Root_Class_3_CA_2_EV_2009.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/d6325660.0 to 'COMODO_RSA_Certification_Authority.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/d7e8dc79.0 to 'QuoVadis_Root_CA_2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/d853d49e.0 to 'Trustis_FPS_Root_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/dc4d6a89.0 to 'GlobalSign_Root_CA_-_R6.crt': Read-only file system.
root@08-00-27-E4-9C-09:/tmp# opkg install openwisp-config_0.6.0a-1_all.ipk 
Installing openwisp-config (0.6.0a-1) to root...
Installing ca-certificates (20200601-1) to root...
Downloading http://downloads.openwrt.org/releases/19.07.2/packages/x86_64/base/ca-certificates_20200601-1_all.ipk
Collected errors:
 * wfopen: //usr/lib/opkg/info/ca-certificates.control: Read-only file system.
 * wfopen: //usr/lib/opkg/info/ca-certificates.postinst: Read-only file system.
 * wfopen: //usr/lib/opkg/info/ca-certificates.prerm: Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/02265526.0 to 'Entrust_Root_Certification_Authority_-_G2.crt': File exists.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/03179a64.0 to 'Staat_der_Nederlanden_EV_Root_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/062cdee6.0 to 'GlobalSign_Root_CA_-_R3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/064e0aa9.0 to 'QuoVadis_Root_CA_2_G3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/06dc52d5.0 to 'SSL.com_EV_Root_Certification_Authority_RSA_R2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/080911ac.0 to 'QuoVadis_Root_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/09789157.0 to 'Starfield_Services_Root_Certificate_Authority_-_G2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/0a775a30.0 to 'GTS_Root_R3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/0b1b94ef.0 to 'CFCA_EV_ROOT.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/0bf05006.0 to 'SSL.com_Root_Certification_Authority_ECC.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/0c4c9b6c.0 to 'Global_Chambersign_Root_-_2008.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/0f5dc4f3.0 to 'UCA_Extended_Validation_Root.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/0f6fa695.0 to 'GDCA_TrustAUTH_R5_ROOT.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/1001acf7.0 to 'GTS_Root_R1.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/106f3e4d.0 to 'Entrust_Root_Certification_Authority_-_EC1.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/128805a3.0 to 'EE_Certification_Centre_Root_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/14bc7599.0 to 'emSign_ECC_Root_CA_-_G3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/1636090b.0 to 'Hellenic_Academic_and_Research_Institutions_RootCA_2011.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/18856ac4.0 to 'SecureSign_RootCA11.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/1d3472b9.0 to 'GlobalSign_ECC_Root_CA_-_R5.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/1e08bfd1.0 to 'IdenTrust_Public_Sector_Root_CA_1.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/1e09d511.0 to 'T-TeleSec_GlobalRoot_Class_2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/244b5494.0 to 'DigiCert_High_Assurance_EV_Root_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/2923b3f9.0 to 'emSign_Root_CA_-_G1.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/2ae6433e.0 to 'CA_Disig_Root_R2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/2b349938.0 to 'AffirmTrust_Commercial.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/2e5ac55d.0 to 'DST_Root_CA_X3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/32888f65.0 to 'Hellenic_Academic_and_Research_Institutions_RootCA_2015.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/349f2832.0 to 'EC-ACC.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/3513523f.0 to 'DigiCert_Global_Root_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/3bde41ac.0 to 'Autoridad_de_Certificacion_Firmaprofesional_CIF_A62634068.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/3e44d2f7.0 to 'TrustCor_RootCert_CA-2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/3e45d192.0 to 'Hongkong_Post_Root_CA_1.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/40193066.0 to 'Certum_Trusted_Network_CA_2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/4042bcee.0 to 'ISRG_Root_X1.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/40547a79.0 to 'COMODO_Certification_Authority.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/406c9bb1.0 to 'emSign_Root_CA_-_C1.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/4304c5e5.0 to 'Network_Solutions_Certificate_Authority.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/48bec511.0 to 'Certum_Trusted_Network_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/4a6481c9.0 to 'GlobalSign_Root_CA_-_R2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/4b718d9b.0 to 'emSign_ECC_Root_CA_-_C3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/4bfab552.0 to 'Starfield_Root_Certificate_Authority_-_G2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/4f316efb.0 to 'SwissSign_Gold_CA_-_G2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/5273a94c.0 to 'E-Tugra_Certification_Authority.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/5443e9e3.0 to 'T-TeleSec_GlobalRoot_Class_3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/54657681.0 to 'Buypass_Class_2_Root_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/57bcb2da.0 to 'SwissSign_Silver_CA_-_G2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/5a4d6896.0 to 'Staat_der_Nederlanden_Root_CA_-_G3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/5ad8a5d6.0 to 'GlobalSign_Root_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/5c44d531.0 to 'Staat_der_Nederlanden_Root_CA_-_G2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/5cd81ad7.0 to 'TeliaSonera_Root_CA_v1.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/5d3033c5.0 to 'TrustCor_RootCert_CA-1.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/5e98733a.0 to 'Entrust_Root_Certification_Authority_-_G4.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/5f15c80c.0 to 'TWCA_Global_Root_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/607986c7.0 to 'DigiCert_Global_Root_G2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/626dceaf.0 to 'GTS_Root_R2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/6410666e.0 to 'Taiwan_GRCA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/653b494a.0 to 'Baltimore_CyberTrust_Root.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/68dd7389.0 to 'Hongkong_Post_Root_CA_3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/6b99d060.0 to 'Entrust_Root_Certification_Authority.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/6d41d539.0 to 'Amazon_Root_CA_2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/6fa5da56.0 to 'SSL.com_Root_Certification_Authority_RSA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/706f604c.0 to 'XRamp_Global_CA_Root.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/749e9e03.0 to 'QuoVadis_Root_CA_1_G3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/75d1b2ed.0 to 'DigiCert_Trusted_Root_G4.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/76cb8f92.0 to 'Cybertrust_Global_Root.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/76faf6c0.0 to 'QuoVadis_Root_CA_3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/7719f463.0 to 'Hellenic_Academic_and_Research_Institutions_ECC_RootCA_2015.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/773e07ad.0 to 'OISTE_WISeKey_Global_Root_GC_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/7aaf71c0.0 to 'TrustCor_ECA-1.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/7f3d5d1d.0 to 'DigiCert_Assured_ID_Root_G3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/8160b96c.0 to 'Microsec_e-Szigno_Root_CA_2009.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/8867006a.0 to 'GeoTrust_Universal_CA_2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/8cb5ee0f.0 to 'Amazon_Root_CA_3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/8d86cdd1.0 to 'certSIGN_ROOT_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/930ac5d2.0 to 'Actalis_Authentication_Root_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/93bc0acc.0 to 'AffirmTrust_Networking.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/988a38cb.0 to 'NetLock_Arany_=Class_Gold=_Főtanúsítvány.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/9c2e7d30.0 to 'Sonera_Class_2_Root_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/9c8dfbd4.0 to 'AffirmTrust_Premium_ECC.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/9d04f354.0 to 'DigiCert_Assured_ID_Root_G2.crt': Read-only file system.
 * wfopen: /etc/ssl/certs/ACCVRAIZ1.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/AC_RAIZ_FNMT-RCM.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Actalis_Authentication_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/AffirmTrust_Commercial.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/AffirmTrust_Networking.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/AffirmTrust_Premium.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/AffirmTrust_Premium_ECC.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Amazon_Root_CA_1.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Amazon_Root_CA_2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Amazon_Root_CA_3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Amazon_Root_CA_4.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Atos_TrustedRoot_2011.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Autoridad_de_Certificacion_Firmaprofesional_CIF_A62634068.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Baltimore_CyberTrust_Root.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Buypass_Class_2_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Buypass_Class_3_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/CA_Disig_Root_R2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/CFCA_EV_ROOT.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/COMODO_Certification_Authority.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/COMODO_ECC_Certification_Authority.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/COMODO_RSA_Certification_Authority.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Certigna.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Certigna_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Certum_Trusted_Network_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Certum_Trusted_Network_CA_2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Chambers_of_Commerce_Root_-_2008.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Comodo_AAA_Services_root.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Cybertrust_Global_Root.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/D-TRUST_Root_Class_3_CA_2_2009.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/D-TRUST_Root_Class_3_CA_2_EV_2009.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/DST_Root_CA_X3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/DigiCert_Assured_ID_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/DigiCert_Assured_ID_Root_G2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/DigiCert_Assured_ID_Root_G3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/DigiCert_Global_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/DigiCert_Global_Root_G2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/DigiCert_Global_Root_G3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/DigiCert_High_Assurance_EV_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/DigiCert_Trusted_Root_G4.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/E-Tugra_Certification_Authority.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/EC-ACC.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/EE_Certification_Centre_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Entrust.net_Premium_2048_Secure_Server_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Entrust_Root_Certification_Authority.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Entrust_Root_Certification_Authority_-_EC1.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Entrust_Root_Certification_Authority_-_G2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Entrust_Root_Certification_Authority_-_G4.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/GDCA_TrustAUTH_R5_ROOT.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/GTS_Root_R1.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/GTS_Root_R2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/GTS_Root_R3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/GTS_Root_R4.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/GeoTrust_Universal_CA_2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/GlobalSign_ECC_Root_CA_-_R4.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/GlobalSign_ECC_Root_CA_-_R5.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/GlobalSign_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/GlobalSign_Root_CA_-_R2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/GlobalSign_Root_CA_-_R3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/GlobalSign_Root_CA_-_R6.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Global_Chambersign_Root_-_2008.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Go_Daddy_Class_2_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Go_Daddy_Root_Certificate_Authority_-_G2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Hellenic_Academic_and_Research_Institutions_ECC_RootCA_2015.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Hellenic_Academic_and_Research_Institutions_RootCA_2011.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Hellenic_Academic_and_Research_Institutions_RootCA_2015.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Hongkong_Post_Root_CA_1.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Hongkong_Post_Root_CA_3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/ISRG_Root_X1.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/IdenTrust_Commercial_Root_CA_1.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/IdenTrust_Public_Sector_Root_CA_1.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Izenpe.com.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/LuxTrust_Global_Root_2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Microsec_e-Szigno_Root_CA_2009.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/NetLock_Arany_=Class_Gold=_Főtanúsítvány.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Network_Solutions_Certificate_Authority.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/OISTE_WISeKey_Global_Root_GA_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/OISTE_WISeKey_Global_Root_GB_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/OISTE_WISeKey_Global_Root_GC_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/QuoVadis_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/QuoVadis_Root_CA_1_G3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/QuoVadis_Root_CA_2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/QuoVadis_Root_CA_2_G3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/QuoVadis_Root_CA_3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/QuoVadis_Root_CA_3_G3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/SSL.com_EV_Root_Certification_Authority_ECC.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/SSL.com_EV_Root_Certification_Authority_RSA_R2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/SSL.com_Root_Certification_Authority_ECC.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/SSL.com_Root_Certification_Authority_RSA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/SZAFIR_ROOT_CA2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/SecureSign_RootCA11.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/SecureTrust_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Secure_Global_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Security_Communication_RootCA2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Security_Communication_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Sonera_Class_2_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Staat_der_Nederlanden_EV_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Staat_der_Nederlanden_Root_CA_-_G2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Staat_der_Nederlanden_Root_CA_-_G3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Starfield_Class_2_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Starfield_Root_Certificate_Authority_-_G2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Starfield_Services_Root_Certificate_Authority_-_G2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/SwissSign_Gold_CA_-_G2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/SwissSign_Silver_CA_-_G2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/T-TeleSec_GlobalRoot_Class_2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/T-TeleSec_GlobalRoot_Class_3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/TUBITAK_Kamu_SM_SSL_Kok_Sertifikasi_-_Surum_1.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/TWCA_Global_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/TWCA_Root_Certification_Authority.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Taiwan_GRCA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/TeliaSonera_Root_CA_v1.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/TrustCor_ECA-1.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/TrustCor_RootCert_CA-1.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/TrustCor_RootCert_CA-2.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Trustis_FPS_Root_CA.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/UCA_Extended_Validation_Root.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/UCA_Global_G2_Root.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/USERTrust_ECC_Certification_Authority.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/USERTrust_RSA_Certification_Authority.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/Verisign_Class_3_Public_Primary_Certification_Authority_-_G3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/XRamp_Global_CA_Root.crt: Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/a3418fda.0 to 'GTS_Root_R4.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/a94d09e5.0 to 'ACCVRAIZ1.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/aee5f10d.0 to 'Entrust.net_Premium_2048_Secure_Server_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/b0e59380.0 to 'GlobalSign_ECC_Root_CA_-_R4.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/b1159c4c.0 to 'DigiCert_Assured_ID_Root_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/b1b8a7f3.0 to 'OISTE_WISeKey_Global_Root_GA_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/b66938e9.0 to 'Secure_Global_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/b727005e.0 to 'AffirmTrust_Premium.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/b7a5b843.0 to 'TWCA_Root_Certification_Authority.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/c01eb047.0 to 'UCA_Global_G2_Root.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/c0ff1f52.0 to 'Verisign_Class_3_Public_Primary_Certification_Authority_-_G3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/c28a8a30.0 to 'D-TRUST_Root_Class_3_CA_2_2009.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/c47d9980.0 to 'Chambers_of_Commerce_Root_-_2008.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/ca6e4ad9.0 to 'ePKI_Root_Certification_Authority.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/cbf06781.0 to 'Go_Daddy_Root_Certificate_Authority_-_G2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/cc450945.0 to 'Izenpe.com.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/cd58d51e.0 to 'Security_Communication_RootCA2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/cd8c0d63.0 to 'AC_RAIZ_FNMT-RCM.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/ce5e74ef.0 to 'Amazon_Root_CA_1.crt': Read-only file system.
 * wfopen: /etc/ssl/certs/certSIGN_ROOT_CA.crt: Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/d4dae3dd.0 to 'D-TRUST_Root_Class_3_CA_2_EV_2009.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/d6325660.0 to 'COMODO_RSA_Certification_Authority.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/d7e8dc79.0 to 'QuoVadis_Root_CA_2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/d853d49e.0 to 'Trustis_FPS_Root_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/dc4d6a89.0 to 'GlobalSign_Root_CA_-_R6.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/dd8e9d41.0 to 'DigiCert_Global_Root_G3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/de6d66f3.0 to 'Amazon_Root_CA_4.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/def36a68.0 to 'LuxTrust_Global_Root_2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/e113c810.0 to 'Certigna.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/e18bfb83.0 to 'QuoVadis_Root_CA_3_G3.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/e36a6752.0 to 'Atos_TrustedRoot_2011.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/e73d606e.0 to 'OISTE_WISeKey_Global_Root_GB_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/e8de2f56.0 to 'Buypass_Class_3_Root_CA.crt': Read-only file system.
 * wfopen: /etc/ssl/certs/ePKI_Root_Certification_Authority.crt: Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/ee64a828.0 to 'Comodo_AAA_Services_root.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/eed8c118.0 to 'COMODO_ECC_Certification_Authority.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/ef954a4e.0 to 'IdenTrust_Commercial_Root_CA_1.crt': Read-only file system.
 * wfopen: /etc/ssl/certs/emSign_ECC_Root_CA_-_C3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/emSign_ECC_Root_CA_-_G3.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/emSign_Root_CA_-_C1.crt: Read-only file system.
 * wfopen: /etc/ssl/certs/emSign_Root_CA_-_G1.crt: Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/f081611a.0 to 'Go_Daddy_Class_2_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/f0c70a8d.0 to 'SSL.com_EV_Root_Certification_Authority_ECC.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/f30dd6ad.0 to 'USERTrust_ECC_Certification_Authority.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/f3377b1b.0 to 'Security_Communication_Root_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/f387163d.0 to 'Starfield_Class_2_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/f39fc864.0 to 'SecureTrust_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/f51bb24c.0 to 'Certigna_Root_CA.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/fc5a8f99.0 to 'USERTrust_RSA_Certification_Authority.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/fe8a2cd8.0 to 'SZAFIR_ROOT_CA2.crt': Read-only file system.
 * extract_archive: Cannot create symlink from ./etc/ssl/certs/ff34af3f.0 to 'TUBITAK_Kamu_SM_SSL_Kok_Sertifikasi_-_Surum_1.crt': Read-only file system.
 * pkg_write_filelist: Failed to open //usr/lib/opkg/info/ca-certificates.list: Read-only file system.
 * opkg_install_pkg: Failed to extract data files for ca-certificates. Package debris may remain!
 * opkg_install_cmd: Cannot install package openwisp-config.
</details>

devkapilbansal reviewed (approved):
Looks good to me. Tested on VM and it works perfectly fine

devkapilbansal commented:
> @devkapilbansal the last line doesn't look good?
> 
> > Read-only file system. * opkg_install_pkg: Failed to extract data files for ca-certificates. Package debris may remain! * opkg_install_cmd: Cannot install package openwisp-config.

May be, there were some problems with my VM. I tried it on virgiin OpenWRT and it worked fine

nemesisdesign reviewed (commented):
Looks good! :+1: 
I'm doing a compile test locally to double check.

nemesisdesign commented:
Local testing went well, thank you @feckert @devkapilbansal :+1:
